controllers/tenant_controller.go

```diff
@@ -89,6 +90,7 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
 		if err := r.finalize(ctx, tenant); err != nil {
 			return ctrl.Result{}, fmt.Errorf("failed to finalize: %w", err)
 		}
+		r.removeMetrics(tenant)
```

Comment:
Move `removeMetrics` to before calling `controllerutil.RemoveFinalizer` in `finalize` function.

This is because once `RemoveFinalizer` is called, this code may never be called again.